### PR TITLE
Intel_LLVM compiler macros

### PR DESCRIPTION
**Pull request by napetrov**
# Description
Update intel compiler macros checks with Intel_LLVM case for migration to ICX compiler

### CODE REVIEW

**Comment by Vika-F** [comment]
There is a build error in cpp/oneapi/dal/algo/linear_kernel/detail/compute_ops.cpp
Because that file is compiled without AVX512-specific compiler options, but it has started to use AVX512 intrinsics after adding __INTEL_LLVM_COMPILER macro into cpp/daal/src/externals/service_blas.h

A quick workaround would be to revert the changes in service_blas.h

**Comment by KulikovNikita** [comment]
Should we modify somethink like [this file](https://github.com/oneapi-src/oneDAL/blob/1d13c2af69952a153e6c0cab41e6006c4eddd05c/cpp/oneapi/dal/backend/common.hpp#L28)?

**Comment by napetrov** [comment]
> There is a build error in cpp/oneapi/dal/algo/linear_kernel/detail/compute_ops.cpp Because that file is compiled without AVX512-specific compiler options, but it has started to use AVX512 intrinsics after adding __INTEL_LLVM_COMPILER macro into cpp/daal/src/externals/service_blas.h
> 
> A quick workaround would be to revert the changes in service_blas.h

yes, as fast solution, but thinking about more longterm one. The problem happens during compilation with -fsycl option which in general wasn't intended to be exposed for icl -> icx migration.  However without this define our daal build with just icx wouldn't go that path.  Will try to look on other available defines to cut of sycl mode

**Comment by napetrov** [comment]
@Vika-F  @KulikovNikita  @homksei  @Alexsandruss - let's review and merge this - we have build for both icc and icx working. 

There is question on overall windows icx but i suggest to solve it as separate PR. meanwhile we would be able to validate ICX linux build and run perf for it

File: cpp/daal/include/services/daal_defines.h [hunk 1]
```diff
@@ -28,7 +28,11 @@
 
 #include <cstddef> // for size_t
 
-#if !(defined(_MSC_VER) || defined(__INTEL_COMPILER))
+#if (defined(__INTEL_COMPILER) || defined(__INTEL_LLVM_COMPILER)) && !defined(SYCL_LANGUAGE_VERSION)
```
[the remark this answers is not in the document]
Comment (napetrov) [reply cpp/daal/include/services/daal_defines.h:31]: Those are C++ core related defines so we need to support icx but not icx -fsycl case.

**Comment by napetrov** [comment]
> Should we modify somethink like [this file](https://github.com/oneapi-src/oneDAL/blob/1d13c2af69952a153e6c0cab41e6006c4eddd05c/cpp/oneapi/dal/backend/common.hpp#L28)?

There is mess in graphs - they are using direct includes of DAAL defines 
#include <daal/src/services/service_defines.h>

this need to be cleaned up before one in common_hpp can be set to all intel compilers. Suggest to do this as another round of changes so we can integrate current changes to get rest of content enabled for icx build

**Review by Alexsandruss — APPROVED** [review]

**Pull request merged**